text/0000-tracked-built-ins.md

```diff
@@ -0,0 +1,270 @@
+to add the dependency to the `app` blueprint, and the `addon` blueprint will
+inherit it automatically.
+
+This has the advantage (over including it as an implicit dependency), that
+apps and addons that don't want to use it for some reason can opt out by
+removing the dependency from their `package.json` file.
+
+## How we teach this
+
+The following guide should be added to the Autotracking In-Depth guide in the official guides, and the
+sections on [POJOs][guides-pojos] and [Array][guides-arrays] should be removed.
+
+### Tracked data structures
+
+Generally, you should try to create classes with their tracked properties enumerated
+and decorated with `@tracked`, instead of relying on dynamically created objects.
+In some cases however, there could be a prohibitive number of possible properties,
+or there could be no way to know them in advance.
+In this case, you should use tracked versions of JavaScript's built-in
+Plain Old JavaScript Object (POJO), array or keyed collection (Maps, Sets, WeakMaps, WeakSets)
+provided by `tracked-built-ins` package.
+
+```js
+import {
+  TrackedObject,
+  TrackedArray,
+  TrackedMap,
+  TrackedSet,
+  TrackedWeakMap,
+  TrackedWeakSet,
+} from 'tracked-built-ins';
+```
+
+For example, we could make a scoreboard component that keeps score for an
+arbitrary number of players, and keep track of the score for each player using
+tracked version of JavaScript [Map][global-objects-map].
+
+```js
+// app/components/scoreboard.js
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { TrackedMap } from 'tracked-built-ins';
+
+export default class Scoreboard extends Component {
+  // Mock data for players - this would eventually be replaced with an argument
+  // so players could be passed in
+  players = [
+    { name: 'Zoey' },
+    { name: 'Tomster' },
+  ];
+
+  // Create a map from player -> score, with each player's score starting at 0
+  scores = new TrackedMap(this.players.map(p => [p, 0]));
+
+  @action
+  incrementScore(player) {
+    let currentScore = this.scores.get(player);
+
+    this.scores.set(player, currentScore + 1);
+  }
+
+  <template>
+    {{#each-in this.scores |player score|}}
+      <div>
+        {{player.name}}: {{score}}
+    
+        <button type="button" {{on "click" (fn this.incrementScore player)}}>
+          +
+        </button>
+      </div>
+    {{/each-in}}
+  </template>
+}
+```
+
+Lets lake a look into another example involving a shopping cart, where we keep track
+of the items added to cart using tracked version of JavaScript's [Array][global-objects-array].
+
+```js
+// app/components/scoreboard.js
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { TrackedArray } from 'tracked-built-ins';
+
+export default class Cart extends Component {
+  // Mock data for cart items - this would eventually be replaced with an argument
+  // so items could be passed in
```

Comment:
As above, this comment ends up being unhelpful and misleading in the context of the RFC (and again, not at all your fault!).

```suggestion
```